Framework/Algorithms/src/ReflectometryReductionOne2.cpp

```diff
@@ -789,24 +790,37 @@ MatrixWorkspace_sptr ReflectometryReductionOne2::algorithmicCorrection(
   return corrAlg->getProperty("OutputWorkspace");
 }
 
-/**
-* The input workspace (in wavelength) to convert to Q
-* @param inputWS : the input workspace to convert
+/** Convert a workspace to Q
+*
+* @param inputWS : The input workspace (in wavelength) to convert to Q
 * @return : output workspace in Q
 */
 MatrixWorkspace_sptr
 ReflectometryReductionOne2::convertToQ(MatrixWorkspace_sptr inputWS) {
-
-  // Convert to Q
-  auto convertUnits = this->createChildAlgorithm("ConvertUnits");
-  convertUnits->initialize();
-  convertUnits->setProperty("InputWorkspace", inputWS);
-  convertUnits->setProperty("Target", "MomentumTransfer");
-  convertUnits->setProperty("AlignBins", false);
-  convertUnits->execute();
-  MatrixWorkspace_sptr IvsQ = convertUnits->getProperty("OutputWorkspace");
-
-  return IvsQ;
+  bool const moreThanOneDetector = inputWS->getDetector(0)->nDets() > 1;
```

Comment:
Could we make this more generic and allow for more than one spectrum? In real usage there should only be one, but I think the algorithm in theory allows multiple detector groups which would result in multiple spectra.

Reply:
I would prefer checking and rejecting grouping to more than 1 group. For multiple groups we would need more than 1 `ThetaIn` but my understanding is that each run of an experiment has only one theta.

Reply:
Sure, that's probably better :+1: